### PR TITLE
Add authorize action to PoM plugin - Closes #5945

### LINKED ISSUE #5945: Update Pom plugin to expose endpoint in action and remove http server
### Description

- Update reportMisbehavior plugin http api endpoint to be an action
- Remove http server

### Motivation

- actions should be preferred way of exposing data so that api client can obtain the data

### Acceptance Criteria

Update functional tests to use actions

### PR DESCRIPTION
**Pull request by mitsuaki-u**
### What was the problem?

This PR resolves #5945

### How was it solved?

- Removed HTTP API endpoint for plugin
- Added appropriate action
- Added unit tests

### How was it tested?

`npm run test`

### CODE REVIEW

File: framework-plugins/lisk-framework-report-misbehavior-plugin/src/report_misbehavior_plugin.ts [hunk 1]
```diff
@@ -91,24 +99,61 @@ export class ReportMisbehaviorPlugin extends BasePlugin {
...
+		return {
+			// eslint-disable-next-line @typescript-eslint/require-await
+			authorize: async (action: ActionInfoObject): Promise<{ result: string }> => {
+				const errors = validator.validate(
+					actionParamsSchema,
+					action.params as Record<string, unknown>,
+				);
+
+				if (errors.length) {
+					throw new Error(new LiskValidationError([...errors]).message);
```
Comment (shuse2) [inline framework-plugins/lisk-framework-report-misbehavior-plugin/src/report_misbehavior_plugin.ts:111]:
```suggestion
					throw new LiskValidationError([...errors]);
```

**Review by shuse2 — COMMENTED** [review]
One minor thing but looks good to me =)

File: framework-plugins/lisk-framework-report-misbehavior-plugin/src/report_misbehavior_plugin.ts [hunk 2]
```diff
@@ -91,24 +99,61 @@ export class ReportMisbehaviorPlugin extends BasePlugin {
 
 	// eslint-disable-next-line class-methods-use-this
 	public get actions(): ActionsDefinition {
-		return {};
+		return {
+			// eslint-disable-next-line @typescript-eslint/require-await
+			authorize: async (action: ActionInfoObject): Promise<{ result: string }> => {
```
Comment (ishantiw) [inline framework-plugins/lisk-framework-report-misbehavior-plugin/src/report_misbehavior_plugin.ts:104]: we can remove async here

**Review by ishantiw — APPROVED** [review]
Just a comment otherwise LGTM

File: framework-plugins/lisk-framework-report-misbehavior-plugin/src/report_misbehavior_plugin.ts (hunk 2 above)
Comment (shuse2) [reply framework-plugins/lisk-framework-report-misbehavior-plugin/src/report_misbehavior_plugin.ts:104]: I think the action interface requires to be async

Comment (ishantiw) [reply framework-plugins/lisk-framework-report-misbehavior-plugin/src/report_misbehavior_plugin.ts:104]: oh then I have to also make one of the actions as async in monitor_plugin https://github.com/LiskHQ/lisk-sdk/blob/efda98cfddba88841f21cf4e78959f26fcb7d85d/framework-plugins/lisk-framework-monitor-plugin/src/monitor_plugin.ts#L84

**Review by shuse2 — APPROVED** [review]

**Pull request merged**